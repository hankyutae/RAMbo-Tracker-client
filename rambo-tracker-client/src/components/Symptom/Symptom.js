import React, { Component } from 'react';
import API from '../../services/Api-service';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import './Symptom.css';

class Symptom extends Component {
  state = {
    symptomName : '',
    symptomSeverity : null,
    symptomTime : new Date(),
  }

  handleSymptomSubmit = (e) => {
    e.preventDefault();
    console.log('testing symptomSubmit');

    let userSymptom = e.target['user-symptom'].value;
    API.doFetch(/*'TODO',*/ 'POST', {userSymptom})
    .then(res => {
      //TODO
    })
    .catch(e => console.log(e));
    e.target['user-symptom'].value = '';
  }

  handleTimeChange = date => { 
    this.setState({ 
      symptomTime: date 
    });
  }

  handleSeverityChange = sev => { 
    this.setState({ 
      symptomSeverity: sev.target.value 
    }); 
  }

  render() {
    return(
      <section className='symptom-container'>
        <form onSubmit={(e)=>this.handleSymptomSubmit(e)}>

        <div id='user-input-container'>
          <label htmlFor='user-symptom'>New Symptom</label>
          <br/>
          <input name='symptom' id='user-symptom' type='text' placeholder='bloated..'></input>
        </div>

        <div>
          <label htmlFor='date-select'>When?</label>
            <DatePicker id='date-select' selected={this.state.symptomTime} onChange={this.handleTimeChange} showTimeSelect withPortal dateFormat="Pp" />
        </div>

        <div id='severity-radio'>
            <p>Rate the Severity</p>
          <div className="radio">
            <label><input type="radio" value="option1" 
            checked={this.state.symptomSeverity === 'option1'}
            onChange={this.handleSeverityChange} />1</label>
          </div>
          <div className="radio">
            <label><input type="radio" value="option2" 
            checked={this.state.symptomSeverity === 'option2'}
            onChange={this.handleSeverityChange} />2</label>
          </div>
          <div className="radio">
            <label><input type="radio" value="option3" 
            checked={this.state.symptomSeverity === 'option3'}
            onChange={this.handleSeverityChange} />3</label>
          </div>
          <div className="radio">
            <label><input type="radio" value="option4" 
            checked={this.state.symptomSeverity === 'option4'}
            onChange={this.handleSeverityChange}/>4</label>
          </div>
          <div className="radio">
            <label><input type="radio" value="option5" 
            checked={this.state.symptomSeverity === 'option5'}
            onChange={this.handleSeverityChange}/>5</label>
          </div>
        </div>

          <br/>
          <button>Submit Symptom</button>
          
        </form>
      </section>
    )
  }


  
}

export default Symptom;