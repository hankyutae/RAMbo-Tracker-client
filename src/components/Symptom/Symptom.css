.symptom-container{
/*   border: 5px groove white;
 */  border-radius: 5px;
  background: rgb(255, 255, 255);
  display: inline-flex;
  flex-direction: column;
  justify-content: left;

  
}

#user-input-container{
  padding: 15px;
}

#date-select{
  text-align: center;
}

#severity-radio{
  padding: 5px;
}

#select{
/*   border: 1px solid black; 
 */  padding: 15px;
  
}

#symptom-select{
  border: 1px solid black;
  margin: 10px 0px;
  width: 60%;
}
@media screen and (max-width: 500px){
#select-preexisting{
width: 75%;
}
}

#date{
  
}
#user-symptom{
  background-color:white;
}
input{
  padding: 5px;
  border-radius: 5px;
  border: 1px solid rgb(255, 255, 255)
}

