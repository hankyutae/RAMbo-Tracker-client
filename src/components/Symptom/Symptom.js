import React, { Component } from "react";
import API from '../../services/api-service';
import DatePicker from "react-datepicker";
import helper from "../../services/helper.services";
import "react-datepicker/dist/react-datepicker.css";
import "./Symptom.css";
let symptomName = document.getElementById('user-symptom')
class Symptom extends Component {
  state = {
    symptomName: "",
    symptomSeverity: 4,
    symptomTime: new Date(),
    pastUserSymptoms: [],
    pastSymptomVal: "",
    symptomSelectIsHidden: false,
    symptomInputDisabled: false,
    error: null
  };

  componentDidMount() {
    this.setState({error: null})
    /* Api.doFetch('TODO')
    .then(res => { */
    //})
    this.setState({ pastUserSymptoms: helper.preExisting() });
    //})
    //.catch(e => console.log(e))
  }


  handleSymptomSubmit = e => {
    e.preventDefault();
    this.setState({error: null})

    let sym = {};
    sym.type='symptom';
    sym.symptom = symptomName.value;
    sym.severity = this.state.symptomSeverity;
    sym.time = this.state.symptomTime;
    sym.symptom = this.state.pastSymptomVal ? this.state.pastSymptomVal : sym.symptom;
    API.doFetch('/event','POST',sym).then(res=>{
      /* res.name = res.type; */
      console.log(res);
      this.props.updateEvents(res);
      this.props.closeModal('addSymptomsModal');//this functions is passed in from dashboard to close the modal, it should be placed int the 'then' of api call to ensure it only runs in happy case 
     }).catch((res)=> {
/*      console.log(res)
 */     this.setState({error: res.error})}); 
  };

  handleTimeChange = date => {
    this.setState({
      symptomTime: date
    });
  };

  handleSeverityChange = sev => {
    console.log(sev.target.value);
    this.setState({
      symptomSeverity: sev.target.value
    });
  };

  handleSymptomChange = sym => {
    this.setState({
      pastSymptomVal: sym.target.value
    });
  };

  addSymptomClick = e => {
    e.preventDefault();
    const symptomSelectIsHidden = this.state.symptomSelectIsHidden;
    this.setState({
      symptomSelectIsHidden: !symptomSelectIsHidden,
      symptomInputDisabled: true
    });
  };

  render() {
    let savedSymptoms = this.state.pastUserSymptoms.map((item, index) => {
      return (
        <option key={index} value={item.value}>
          {item.label}
        </option>
      );
    });

    return (
      <div className="symptom-container">
      <section className="symptom-container">
        <h2>Log a Symptom</h2>
{/*         {this.state.error && <p className="error">There Was An Error</p>}
 */}        <form onSubmit={e => this.handleSymptomSubmit(e)}>
          <div id="user-input-container">
            <label htmlFor="user-symptom">Add New Symptom</label>
            <br />
            <datalist id="past-symptoms">
              {this.state.pastUserSymptoms.map((sym,i)=><option key = {i} value={sym.label}/>)}
            </datalist>
            <input
              name="symptom"
              onFocus = {(e)=>symptomName = e.target}
              id="user-symptom"
              type="text"
              placeholder="bloated.."
              list= "past-symptoms"
              disabled={this.state.symptomSelectIsHidden}
            />
            <p id="auto-complete"></p>
          </div>
    {this.props.prevSymptoms.map((s,i)=>{
    if(i > 4)//limits to 5
      return null;
    return <input type='button' key={i} onFocus={(e)=>symptomName = e.target} name="pastSymptoms" value={s.name}/>
    })
    }

          {/* 
            <button id="select-preexisting" className='user-button' onClick={e => this.addSymptomClick(e)}>
              Choose Saved Symptom
            </button>
            {this.state.symptomSelectIsHidden ? (
              <>
                <label htmlFor="symptom-select"></label>
                <select
                  id="symptom-select"
                  onChange={this.handleSymptomChange}
                  value={this.state.pastSymptomVal}
                >
                  {savedSymptoms}
                </select>{" "}
              </>
            ) : (
              <></>
            )} */}

          <div id="date">
            <label htmlFor="date-select">Date and Time:</label>
            <DatePicker
              id="date-select"
              selected={this.state.symptomTime}
              onChange={this.handleTimeChange}
              showTimeSelect
              withPortal
              dateFormat="Pp"
            />
          </div>

          <div id="severity-radio">
            <label 
              htmlFor='severity-slider'>
              Rate the Severity
            </label><br/>
            
            <input
              id='severity-slider'
              type="range"
              step="1"
              min="1"
              max="5"
              list="tickmarks"
              defaultValue={this.state.symptomSeverity}
              onChange={e => this.handleSeverityChange(e)}
            />
            <datalist id="tickmarks">
              <option value="1" label="low"></option>
              <option value="2"></option>
              <option value="3" label="3"></option>
              <option value="4"></option>
              <option value="5" label="5"></option>
            </datalist>
            <p id="severity-desc">(Scale of 1-5: 1 being low, 5 being extreme)</p>
          </div>
          <br />
          {this.state.error && <p className='symptom-error'>{this.state.error}</p>}
          <div id='submit-button'>
            <button className="user-button" type="submit">
              Submit Symptom
            </button>
          </div>
        </form>
      </section>
      </div>
    );
  }
}

export default Symptom;
