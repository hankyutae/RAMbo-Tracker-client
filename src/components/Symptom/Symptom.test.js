import React from "react";
import ReactDOM from "react-dom";
import Symptom from "./Symptom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

const props = {
  prevSymptoms: ["a", "b", "c"]
};
it("renders without crashing", () => {
  const form = document.createElement("form");
  const radio = document.createElement("input");
  form.id = "symptom-form";
  radio.name = "radio-face";
  radio.type = "radio";
  form.appendChild(radio);
  const div = document.createElement("div");
  div.appendChild(form)
  ReactDOM.render(
    <BrowserRouter>
      <Symptom {...props}/>
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Symptom {...props} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
