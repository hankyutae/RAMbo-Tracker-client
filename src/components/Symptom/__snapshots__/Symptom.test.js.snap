// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="symptom-container"
>
  <h2>
    Log a Symptom
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      id="user-input-container"
    >
      <label
        htmlFor="user-symptom"
      >
        Add New Symptom
      </label>
      <br />
      <input
        disabled={false}
        id="user-symptom"
        name="symptom"
        placeholder="bloated.."
        type="text"
      />
    </div>
    <div
      id="select"
    >
      <button
        className="user-button"
        id="select-preexisting"
        onClick={[Function]}
      >
        Choose Saved Symptom
      </button>
    </div>
    <div
      id="date"
    >
      <label
        htmlFor="date-select"
      >
        Date and Time:
      </label>
      <div>
        <div
          className="react-datepicker__input-container"
        >
          <input
            className=""
            disabled={false}
            id="date-select"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={false}
            type="text"
            value="09/24/2019, 11:53 AM"
          />
        </div>
      </div>
    </div>
    <div
      id="severity-radio"
    >
      <label
        htmlFor="severity-slider"
      >
        Rate the Severity
      </label>
      <br />
      <input
        defaultValue={4}
        id="severity-slider"
        max="5"
        min="1"
        onChange={[Function]}
        step="1"
        type="range"
      />
    </div>
    <br />
    <div
      id="submit-button"
    >
      <button
        className="user-button"
        type="submit"
      >
        Submit Symptom
      </button>
    </div>
  </form>
</section>
`;
